src/porepy/numerics/fv/mpfa.py

```diff
@@ -921,7 +921,10 @@ def _flux_discretization(
         # unknown in the discretized sytsem to the right hand side.
         # The negative in front of pr_trace_cell comes from the grad_eqs
         rhs_cells = -sps.vstack([nk_cell, -pr_trace_cell, pr_cont_cell])
-        flux = darcy * igrad * rhs_cells
+        # Compute matrix-matrix product, this is used in several steps below
+        darcy_igrad = darcy * igrad
+
+        flux = darcy_igrad * rhs_cells
```

Comment:
Could this decomposition help us in differentiable tpfa? It seems that you've found an improved factorisation, which might be exactly what we want.